code/modules/research/xenobiology/crossbreeding/warping_runes.dm

```diff
@@ -0,0 +1,309 @@
+///////////////////////////////////////////////
+//////////////////SLIME RUNES//////////////////
+///////////////////////////////////////////////
+
+/obj/effect/slimerune/grey
+	name = "grey rune"
+	desc = "The center has an outline of a slime extract in it."
+
+/obj/effect/slimerune/grey/examine(mob/user)
+	..()
+	var/obj/item/slimecross/warping/grey/ex = extract
+	to_chat(user, "<span class='notice'>It has absorbed [ex.num_absorbed] extract[(ex.num_absorbed != 1) ? "s":""].</span>")
+	to_chat(user, "<span class='notice'>At 8 extracts, it can be manually fed an extract to produce a slime of that color.</span>")
+
+/obj/effect/slimerune/grey/process()
+	var/obj/item/slimecross/warping/grey/ex = extract
+	if(ex.num_absorbed >= 8)
+		return
+	if(locate(/obj/item/slime_extract) in get_turf(loc))
```

Comment:
You can skip the locate and go straight to the for loop. Cancel after if none are eaten